example/index.js

```diff
@@ -85,23 +97,24 @@ class GoogleSigninSampleApp extends Component {
 
   _signIn = async () => {
     try {
-      const user = await GoogleSignin.signIn();
-      this.setState({ user, error: null });
+      const userInfo = await GoogleSignin.signIn();
+      this.setState({ userInfo, error: null });
     } catch (error) {
-      if (error.code === 'CANCELED') {
-        error.message = 'user canceled the login flow';
+      if (doesErrorNeedToBeHandled(error)) {
```

Comment:
idk, i might be wrong so I'd love hear more arguments to this, but maybe this should be handled by the library, it shouldn't throw the error if it doesn't need to be handled.

Reply:
@AndreiCalazans so, if you call `signIn`, two things can happen 

- the promise gets resolved (which always means that the user signed in and you can read their details)
- the promise gets rejected, which can have many causes. Among these causes are login cancellation (ie. user clicks cancel at any point of the process) or situations where some async operation is already in progress (eg. user taps the sign in button twice in row real fast - the first tap starts opening the sign in modal, the second tap gets rejected because the operation is already in progress). When working with the sign in promise, people may put a `Alert.alert()` call into the `catch` block. That means if the user cancels the login, they will get an alert. Or if they tap the login button fast several times, they will get another alert. However, these are not the kind of alerts that tell the user something valuable - the user will just think that the app is buggy. This is why I included the `doesErrorNeedToBeHandled` function - it determines if the error represents login cancellation and situations when some operation is already in progress. Does this make sense? I'm ofc open to discussing the best approach.